Adds cudn docs (#85165)
openshift/openshift-docs: modules/nw-cudn-best-practices.adoc

//module included in the following assembly:
//
// * networking/multiple_networks/primary_networks/about-user-defined-networks.adoc

:_mod-docs-content-type: CONCEPT
[id="considerations-for-cudn_{context}"]
= Best practices for ClusterUserDefinedNetwork CRs

Before setting up a `ClusterUserDefinedNetwork` custom resource (CR), users should consider the following information:

* A `ClusterUserDefinedNetwork` CR is intended for use by cluster administrators and should not be used by non-administrators. If used incorrectly, it might result in security issues with your deployment, cause disruptions, or break the cluster network. 

* `ClusterUserDefinedNetwork` CRs should not be created in the default namespace.

* `ClusterUserDefinedNetwork` CRs should not target `openshift-*` namespaces.

> * {product-title} administrators should be aware that empty an `matchLabel` selector results in selecting all namespaces of the cluster.

Author: Addressed: 

* {product-title} administrators should be aware that all namespaces of a cluster are selected when the following conditions are met:

** The `matchLabels` selector is left empty.
** The `matchExpressions` selector is left empty.
** The `namespaceSelector` is initialized, but does not specify `matchExpressions` or `matchLabel`. For example: `namespaceSelector: {}`.

> * {product-title} administrators should be aware that empty `matchLabel` selectors results in selecting all namespaces of the cluster.

Reviewer: ```suggestion
* {product-title} administrators should be aware that an empty `matchLabel` selector results in selecting all namespaces of the cluster.
```

> * For primary networks, the namespace used for the `ClusterUserDefinedNetwork` CR must include the `k8s.ovn.org/primary-user-defined-network` label. This label cannot be updated, and can only be added when the namespace is created. The following conditions apply with the `k8s.ovn.org/primary-user-defined-network` namespace label:

Reviewer 2: Please note this is true for primary UserDefinedNetwork CRs as well.
It should probably addressed in follow-up PR if we dont have it already.

Author: Yep--it's documented in this PR: https://github.com/openshift/openshift-docs/pull/86845/files#diff-a4571cfec54163a4dbb46db34db71f3ea359a229f8c22eb0413727622160a934R20

** If the namespace is missing the `k8s.ovn.org/primary-user-defined-network` label and a pod is created, the pod attaches itself to the default network.

> ** If the namespace is missing the `k8s.ovn.org/primary-user-defined-network` label and a primary CUDN CR is created that matches the namespace, the CUDN reports an error status and the network will not be created.

Reviewer: ```suggestion
** If the namespace is missing the `k8s.ovn.org/primary-user-defined-network` label and a primary CUDN CR is created that matches the namespace, the CUDN reports an error status and the network is not created.
```

> ** If the namespace is missing the `k8s.ovn.org/primary-user-defined-network` label and a primary CUDN already exists, a pod in the namespace will be created and attached to the default network.

Reviewer: ```suggestion
** If the namespace is missing the `k8s.ovn.org/primary-user-defined-network` label and a primary CUDN already exists, a pod in the namespace is created and attached to the default network.
```

> ** If the namespace _has_ the label, and a primary CUDN does not exist, a pod in the namespace will not be created until the CUDN is created.

Reviewer: ```suggestion
** If the namespace _has_ the label, and a primary CUDN does not exist, a pod in the namespace is not created until the CUDN is created.
```
